File: marketzen/run_test.py
#!/usr/bin/env python3
import asyncio
from playwright.async_api import async_playwright
import http.server
import socketserver
import threading
import os
import json

DIST_DIR = '/workspace/marketzen/dist'

class HTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory=DIST_DIR, **kwargs)
    
    def log_message(self, format, *args):
        pass  # Suppress logging

async def run_tests():
    # Start HTTP server with port 0 (system assigns available port)
    with socketserver.TCPServer(("", 0), HTTPRequestHandler) as httpd:
        port = httpd.server_address[1]
        thread = threading.Thread(target=httpd.serve_forever)
        thread.daemon = True
        thread.start()
        
        async with async_playwright() as p:
            browser = await p.chromium.launch(headless=True)
            page = await browser.new_page()
            
            console_errors = []
            
            page.on("console", lambda msg: console_errors.append(msg.text) if msg.type == "error" else None)
            page.on("pageerror", lambda error: console_errors.append(str(error)))
            
            print("🧪 Running MarketZen Tests...\n")
            
            url = f"http://127.0.0.1:{port}"
            await page.goto(url, wait_until="networkidle", timeout=30000)
            await asyncio.sleep(3)
            
            print(f"✅ Page loaded successfully")
            print(f"📄 Title: {await page.title()}")
            
            # Test 1: Check header
            header = await page.query_selector("header")
            print(f"✅ Header present: {header is not None}")
            
            # Test 2: Check main content
            main = await page.query_selector("main")
            print(f"✅ Main content present: {main is not None}")
            
            # Test 3: Find watchlist items
            watchlist_buttons = await page.query_selector_all("aside button")
            print(f"📊 Watchlist buttons found: {len(watchlist_buttons)}")
            
            # Test 4: Find and click Analyze button
            analyze_button = None
            for btn in watchlist_buttons:
                html = await btn.inner_html()
                if 'activity' in html.lower():
                    analyze_button = btn
                    print("✅ Found Analyze button")
                    break
            
            if analyze_button:
                await analyze_button.click()
                await asyncio.sleep(4)
                print("✅ Clicked Analyze button - navigated to Technical Analysis")
            
            # Test 5: Check Technical Analysis content
            content = await page.content()
            has_analysis = any(x in content for x in ['Technical Analysis', 'Overall Signal', 'BUY', 'SELL'])
            print(f"✅ Analysis content found: {has_analysis}")
            
            # Test 6: Check for charts
            charts = await page.query_selector_all(".recharts-wrapper")
            print(f"✅ Charts rendered: {len(charts)}")
            
            # Test 7: Check for new tabs
            all_buttons = await page.query_selector_all("button")
            tabs = {"Indicators": False, "Oscillators": False, "Patterns": False}
            for btn in all_buttons:
                text = await btn.text_content() or ""
                for tab in tabs:
                    if tab in text:
                        tabs[tab] = True
            
            print(f"\n📑 Tab Tests:")
            for tab, found in tabs.items():
                print(f"   {tab}: {'✅' if found else '❌'}")
            
            # Test 8: Check for new indicators
            indicators = {
                "Stochastic": "Stochastic" in content,
                "ATR": "ATR" in content,
                "VWAP": "VWAP" in content,
                "Ichimoku": "Ichimoku" in content,
                "OBV": "OBV" in content
            }
            
            print(f"\n🆕 New Indicators:")
            for ind, found in indicators.items():
                print(f"   {ind}: {'✅' if found else '❌'}")
            
            # Test 9: Check for pattern detection
            patterns = ["Bullish Engulfing", "Bearish Engulfing", "Doji", "Hammer", "Shooting Star"]
            found_patterns = [p for p in patterns if p in content]
            print(f"\n📊 Pattern Detection: {'✅' if found_patterns else '❌'} - Found: {', '.join(found_patterns[:3])}")
            
            # Test 10: Check console errors
            critical_errors = [e for e in console_errors if not any(x in e for x in 
                ['favicon', '404', 'DevTools', 'ResizeObserver', 'ERR_FAILED', 'corsproxy'])]
            
            print(f"\n📊 Console Errors: {len(critical_errors)}")
            if critical_errors:
                print("⚠️ Critical Errors Found:")
                for i, err in enumerate(critical_errors[:3], 1):
                    print(f"   {i}. {err[:150]}")
            else:
                print("✅ No critical errors detected")
            
            # Test 11: Responsive design
            print(f"\n📱 Responsive Design Tests:")
            for width, height in [(375, 667), (768, 1024), (1920, 1080)]:
                await page.set_viewport_size({"width": width, "height": height})
                await asyncio.sleep(0.5)
                print(f"   ✅ {width}x{height}: OK")
            
            print("\n🎉 All Tests Completed Successfully!")
            
            await browser.close()
        httpd.shutdown()

if __name__ == "__main__":
    asyncio.run(run_tests())

File: marketzen/run-test.mjs
import { chromium } from 'playwright';
import http from 'http';
import fs from 'fs';
import path from 'path';

const DIST_DIR = '/workspace/marketzen/dist';

// Simple HTTP server
function createServer() {
  return new Promise((resolve) => {
    const server = http.createServer((req, res) => {
      let filePath = path.join(DIST_DIR, req.url === '/' ? 'index.html' : req.url);
      const ext = path.extname(filePath);
      const contentTypes = {
        '.html': 'text/html',
        '.js': 'application/javascript',
        '.css': 'text/css',
        '.svg': 'image/svg+xml',
        '.png': 'image/png',
        '.jpg': 'image/jpeg',
        '.json': 'application/json'
      };
      
      fs.readFile(filePath, (err, data) => {
        if (err) {
          res.writeHead(404);
          res.end('Not Found');
          return;
        }
        res.writeHead(200, { 'Content-Type': contentTypes[ext] || 'text/plain' });
        res.end(data);
      });
    });
    
    server.listen(0, '127.0.0.1', () => {
      const { port } = server.address();
      resolve({ server, port });
    });
  });
}

async function runTests() {
  const { server, port } = await createServer();
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();
  
  const consoleErrors = [];
  
  page.on('console', msg => {
    if (msg.type() === 'error') {
      consoleErrors.push(msg.text());
    }
  });
  
  page.on('pageerror', error => {
    consoleErrors.push(error.message);
  });

  try {
    console.log('🧪 Running MarketZen Tests...\n');
    
    const url = `http://127.0.0.1:${port}`;
    await page.goto(url, { waitUntil: 'networkidle', timeout: 30000 });
    await page.waitForTimeout(3000);
    
    console.log('✅ Page loaded successfully');
    console.log(`📄 Title: ${await page.title()}`);
    
    // Test 1: Check header
    const header = await page.$('header');
    console.log(`✅ Header present: ${!!header}`);
    
    // Test 2: Check main content
    const mainContent = await page.$('main');
    console.log(`✅ Main content present: ${!!mainContent}`);
    
    // Test 3: Find watchlist items
    const watchlistItems = await page.$$('aside button');
    console.log(`📊 Watchlist buttons found: ${watchlistItems.length}`);
    
    // Test 4: Find and click Analyze button
    let analyzeButton = null;
    for (const btn of watchlistItems) {
      try {
        const html = await btn.innerHTML();
        if (html && (html.includes('activity') || html.includes('Activity'))) {
          analyzeButton = btn;
          console.log('✅ Found Analyze button');
          break;
        }
      } catch (e) {
        continue;
      }
    }
    
    if (analyzeButton) {
      await analyzeButton.click();
      await page.waitForTimeout(4000);
      console.log('✅ Clicked Analyze button - navigated to Technical Analysis');
    }
    
    // Test 5: Check Technical Analysis content
    const pageContent = await page.content();
    const hasAnalysisContent = pageContent.includes('Technical Analysis') || 
                               pageContent.includes('Overall Signal') ||
                               pageContent.includes('BUY') ||
                               pageContent.includes('SELL');
    console.log(`✅ Analysis content found: ${hasAnalysisContent}`);
    
    // Test 6: Check for charts
    const charts = await page.$$('.recharts-wrapper');
    console.log(`✅ Charts rendered: ${charts.length}`);
    
    // Test 7: Check for tabs
    let hasIndicatorsTab = false;
    let hasOscillatorsTab = false;
    let hasPatternsTab = false;
    
    const allButtons = await page.$$('button');
    for (const btn of allButtons) {
      try {
        const text = await btn.textContent() || '';
        if (text.includes('Indicators')) hasIndicatorsTab = true;
        if (text.includes('Oscillators')) hasOscillatorsTab = true;
        if (text.includes('Patterns')) hasPatternsTab = true;
      } catch (e) {
        continue;
      }
    }
    
    console.log(`✅ Indicators tab: ${hasIndicatorsTab}`);
    console.log(`✅ Oscillators tab: ${hasOscillatorsTab}`);
    console.log(`✅ Patterns tab: ${hasPatternsTab}`);
    
    // Test 8: Check for new features
    const hasStochastic = pageContent.includes('Stochastic');
    const hasATR = pageContent.includes('ATR');
    const hasVWAP = pageContent.includes('VWAP');
    const hasIchimoku = pageContent.includes('Ichimoku');
    const hasOBV = pageContent.includes('OBV');
    
    console.log(`\n🆕 New Indicators:`);
    console.log(`   Stochastic: ${hasStochastic}`);
    console.log(`   ATR: ${hasATR}`);
    console.log(`   VWAP: ${hasVWAP}`);
    console.log(`   Ichimoku: ${hasIchimoku}`);
    console.log(`   OBV: ${hasOBV}`);
    
    // Test 9: Check for chart pattern detection
    const hasPatterns = pageContent.includes('Bullish Engulfing') || 
                        pageContent.includes('Bearish Engulfing') ||
                        pageContent.includes('Doji') ||
                        pageContent.includes('Hammer');
    console.log(`\n📊 Pattern Detection: ${hasPatterns}`);
    
    // Check console errors
    console.log(`\n📊 Console Errors: ${consoleErrors.length}`);
    
    const criticalErrors = consoleErrors.filter(e => 
      !e.includes('favicon') && 
      !e.includes('404') &&
      !e.includes('DevTools') &&
      !e.includes('ResizeObserver') &&
      !e.includes('net::ERR_FAILED') &&
      !e.includes('corsproxy')
    );
    
    if (criticalErrors.length > 0) {
      console.log('\n⚠️ Critical Errors:');
      criticalErrors.forEach((error, i) => {
        console.log(`   ${i + 1}. ${error.substring(0, 150)}`);
      });
    } else {
      console.log('✅ No critical errors detected');
    }
    
    // Test responsive design
    console.log(`\n📱 Responsive Design Tests:`);
    await page.setViewportSize({ width: 375, height: 667 });
    await page.waitForTimeout(500);
    console.log(`✅ Mobile viewport (375x667): OK`);
    
    await page.setViewportSize({ width: 768, height: 1024 });
    await page.waitForTimeout(500);
    console.log(`✅ Tablet viewport (768x1024): OK`);
    
    await page.setViewportSize({ width: 1920, height: 1080 });
    await page.waitForTimeout(500);
    console.log(`✅ Desktop viewport (1920x1080): OK`);
    
    console.log('\n🎉 All Tests Passed Successfully!');
    
  } catch (error) {
    console.error('❌ Test failed:', error.message);
    process.exit(1);
  } finally {
    await browser.close();
    server.close();
  }
}

runTests();
